pkg/portutil/portutil.go

```diff
@@ -18,14 +18,31 @@ package portutil
 
 import (
 	"net"
-	"strconv"
 	"strings"
 
 	gocni "github.com/containerd/go-cni"
+	"github.com/docker/go-connections/nat"
 	"github.com/pkg/errors"
 )
 
-func ParseFlagP(s string) (*gocni.PortMapping, error) {
+func splitParts(rawport string) (string, string, string) {
+	parts := strings.Split(rawport, ":")
+	n := len(parts)
+	containerport := parts[n-1]
+
+	switch n {
+	case 1:
+		return "", "", containerport
+	case 2:
+		return "", parts[0], containerport
+	case 3:
+		return parts[0], parts[1], containerport
+	default:
+		return strings.Join(parts[:n-2], ":"), parts[n-2], containerport
```

Comment:
The switch is hard to read. Is this `default` case for IPv6?

Reply:
Nope, I targeting Ipv4 here

look a this two example :
`./nerdctl --snapshotter native run -d -p 1515:127.0.0.1:80:80 alpine sleep 100` 

parts = [1515 127.0.0.1 80 80]

parts[n-1] -> 80
parts[n-2]  -> 80
parts[:n-2] -> [ 1515 127.0.0.1]  So strings.Join(parts[:n-2], ":") => 1515:127.0.0.1 (original input)

`./nerdctl --snapshotter native run -d -p 1515-127.0.0.1:80:80 alpine sleep 100` 

parts = [1515-127.0.0.1 80 80]

parts[n-1] -> 80
parts[n-2]  -> 80
parts[:n-2] -> [1515-127.0.0.1]  So strings.Join(parts[:n-2], ":") => 1515-127.0.0.1 

it is ok for ipv6 too.

Reply:
I don't understand what `1515-127.0.0.1:80:80` means. Is it officially supported by Docker?

Reply:
@AkihiroSuda  it is a wrong IP that a user may try (just an example)

the purpose is to get  the IP block and check it.